source/common/tcp_proxy/upstream.h

```diff
@@ -44,10 +46,11 @@ class HttpConnPool : public GenericConnPool, public Http::ConnectionPool::Callba
 
   HttpConnPool(const std::string& cluster_name, Upstream::ClusterManager& cluster_manager,
                Upstream::LoadBalancerContext* context, const TunnelingConfig& config,
-               Tcp::ConnectionPool::UpstreamCallbacks& upstream_callbacks);
+               Tcp::ConnectionPool::UpstreamCallbacks& upstream_callbacks,
+               Http::CodecClient::Type type);
   ~HttpConnPool() override;
 
-  bool valid() const { return conn_pool_ != nullptr; }
+  bool valid() const { return conn_pool_ != nullptr && type_ <= Http::CodecClient::Type::HTTP2; }
```

Comment:
Worth a comment to explain why HTTP3 is not considered a valid upstream protocol?

What tests we have to cover the case of type_ set to HTTP3?

Reply:
well, HTTP3 upstreams are not supported yet AFAIK, maybe I should simply remove this test. WDYT?

Reply:
I lean towards keeping the validation, since it future-proofs us from messed up configs and custom upstreams.

Reply:
ok, I will just add a comment saying that this is to make it future-proof then (unless you think I should add a unit-test but I don't see much value in it)